query/sources/impl/src/main/java/org/codice/ddf/admin/sources/opensearch/persist/SaveOpenSearchConfiguration.java

```diff
@@ -0,0 +1,91 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.admin.sources.opensearch.persist;
+
+import static org.codice.ddf.admin.common.message.DefaultMessages.noExistingConfigError;
+import static org.codice.ddf.admin.common.services.ServiceCommons.configExists;
+import static org.codice.ddf.admin.common.services.ServiceCommons.update;
+import static org.codice.ddf.admin.sources.commons.SourceActionCommons.persistSource;
+import static org.codice.ddf.admin.sources.commons.utils.SourceValidationUtils.validateSourceName;
+import static org.codice.ddf.admin.sources.services.OpenSearchServiceProperties.openSearchConfigToServiceProps;
+
+import java.util.List;
+
+import org.apache.commons.lang.StringUtils;
+import org.codice.ddf.admin.api.fields.Field;
+import org.codice.ddf.admin.common.actions.BaseAction;
+import org.codice.ddf.admin.common.fields.base.scalar.BooleanField;
+import org.codice.ddf.admin.configurator.ConfiguratorFactory;
+import org.codice.ddf.admin.sources.fields.ServicePid;
+import org.codice.ddf.admin.sources.fields.type.OpenSearchSourceConfigurationField;
+
+import com.google.common.collect.ImmutableList;
+
+public class SaveOpenSearchConfiguration extends BaseAction<BooleanField> {
+
+    public static final String ID = "saveOpenSearchSource";
+
+    public static final String DESCRIPTION =
+            "Saves an OpenSearch source configuration. If a pid is specified, the source configuration specified by the pid will be updated. Returns true on success and false on failure.";
+
+    private OpenSearchSourceConfigurationField config;
+
+    private ServicePid servicePid;
+
+    private ConfiguratorFactory configuratorFactory;
+
+
+    public SaveOpenSearchConfiguration(ConfiguratorFactory configuratorFactory) {
+        super(ID, DESCRIPTION, new BooleanField());
+        config = new OpenSearchSourceConfigurationField();
+        servicePid = new ServicePid();
+        config.isRequired(true);
```

Comment:
probably easier to use `allInnerFieldsRequired` and state which are not required instead. It seems like it'll be a bit more readable too.